### DISCUSSION TITLE
cordova-plugin-background

### DISCUSSION THREAD

**Discussion by pierrot10** [npm/feedback#281] [CLI] [answered]
Hello
I am new with Cordova 10 and I have just create a Android/IOS App with jQuery Mobile
Each 5mn it need to make a MySQL request to collect some data and update a map. That woks fine but apparently, when the smartphone goes to sleep (I need to unlock) , I receive no alarms.

I need to install a backgound plugin and this on look nice:
cordova-plugin-advanced-background-mode

As I am not very sure how to use it, I would like to have your recommendation

In my javascript, I have the following code
```
document.addEventListener('deviceready', onDeviceReady, false);
function onDeviceReady() {
    document.getElementById('deviceready').classList.add('deviseIsReady');
    if(d_bug == false)
    {
        if(Pages.checkConnection() == true)
        {
            Maps.load(); 
            setInterval(Maps.load, 300000);   
        }
        Pages.checkNotification();
    }
}
```

If the connection is good, Maps.load() send a MySQL and display some stations on the map with temperatures.
Then, each 5mn, and new request is done ans the map is updated.

I supposed first, I have to add the following, when my devise is ready

```
document.addEventListener('deviceready', onDeviceReady, false);
function onDeviceReady() {
    document.getElementById('deviceready').classList.add('deviseIsReady');
    cordova.plugins.backgroundMode.enable();

    if(Pages.checkConnection() == true)
        {
            Maps.load(); 
            setInterval(Maps.load, 300000);   
        }
}
```
I wonder, if it's enough???

My only goal is to make sure my app always send request to the DB and if the temperature is too low, it ring and vibrate the smartphone. If the map us update, that better.

I also looked at this 

```
cordova.plugins.backgroundMode.isActive();
cordova.plugins.backgroundMode.on('EVENT', function); // fire when the mode change
```
I wonder, if it's necessary to use the event and to control if the background is active or not.

When the mode change, do I need to recall Maps.load()?

Is there any example?

Many thanks

**Comment by pierrot10** [comment]
Other things,
I also see that

`cordova.plugins.backgroundMode.moveToBackground();`

Would it be reasonable to do that

```
$(window).load(function(){

    document.addEventListener("offline", Pages.offLine, false);
    document.getElementById("refresh").addEventListener("click", Maps.refresh);
    document.getElementById("ffield").addEventListener("change", Maps.field);
    document.getElementById("fstations").addEventListener("change", Charts.changeStation);
    document.getElementById("threshold").addEventListener("change", Pref.threshold);
    document.getElementById("ffieldpref").addEventListener("change", Pref.field);
    cordova.plugins.backgroundMode.on('EVENT', backgroundEvent);
    
});


function backgroundEvent(){
    if(cordova.plugins.backgroundMode.isActive()){
        cordova.plugins.backgroundMode.moveToBackground();
    }
    else
    {
        cordova.plugins.backgroundMode.moveToForeground();
    }
}
```

If it is a good idea, what will happen for IOS devises?

**Comment by MylesBorins** [comment, marked as answer]
Hey @pierrot10 thanks for opening this discussion.

The npm/feedback repo is specifically for giving suggestions about new feature for our various npm products such as the registry, website, and CLI. We don't offer general technical support via this channel. You may have better luck asking over at https://github.community or stackoverflow.

Best of luck figuring this out.
